source/js/components/petition/petition.jsx

```diff
@@ -584,7 +569,7 @@ export default class Petition extends React.Component {
                 disabled={disableFields}
                 onFocus={this.onInputFocus}
               />
-              {this.state.userTriedSubmitting && (!this.email.element.value || !this.validatesAsEmail(this.email.element.value)) && <small className="form-check form-control-feedback">{ get(`Please enter your email`) }</small>}
```

Comment:
As we are removing the function, but, we still want the email to not be null, hence I left the checking of the element's value.